src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/20.diff.sql

```diff
@@ -0,0 +1,353 @@
+END;
+
+EXEC dbo.LogSchemaMigrationProgress 'Creating initial partitions for resource change data table.';
+GO
+
+IF (EXISTS(SELECT 1 FROM sys.partition_functions WHERE name = 'PartitionFunction_ResourceChangeData_Timestamp')
+    AND EXISTS(SELECT 1 FROM sys.partition_schemes WHERE name = 'PartitionScheme_ResourceChangeData_Timestamp'))
+BEGIN
+    /* Creates initial partitions based on default 48-hour retention period and 1-month future partitions. */
+    DECLARE @numberOfHistoryPartitions int = 48;
+
+    /* To have a buffer time when an error occurs related to partition creation, 
+       by default 720 hours of partitions for the future DateTime will be created in the resource change data table. 
+       The number of future partitions is 720 for 30 days. */
+    DECLARE @numberOfFuturePartitions int = 720;
+    DECLARE @rightPartitionBoundary datetime2(7);
+    DECLARE @currentDateTime datetime2(7) = sysutcdatetime();
+
+    /* There will be 771 partitions, and 770 partition boundaries, one for partition anchor DateTime,
+       48 partition boundaries for history, one for the current hour, and 720 for the future datetimes.
+       Creates 720 partition boudaries for the future to mitigate risk to any data movement
+       and have a buffer time to investigate an issue when an error occurs on partition creation.
+       Once a database is upgraded, a purge change data worker will be run hourly to maintain the number of partitions on resource change datatable. 
+       The partition anchor boundary will be removed at the very first run of the purge operation of the purge change data worker.
+       Total number of partition boundaries = the number of history partitions + one for the current hour + the number of future partitions. */
+    WHILE @numberOfHistoryPartitions >= -@numberOfFuturePartitions
+    BEGIN
+        /* Rounds the start datetime to the hour. */
+        SET @rightPartitionBoundary = DATEADD(hour, DATEDIFF(hour, 0, @currentDateTime) - @numberOfHistoryPartitions, 0);
+
+        /* Checks if a partition exists. */
+        IF NOT EXISTS (SELECT 1 FROM sys.partition_range_values AS prv
+                        JOIN sys.partition_functions AS pf
+                            ON pf.function_id = prv.function_id
+                    WHERE pf.name = N'PartitionFunction_ResourceChangeData_Timestamp'
+                        and CAST(prv.value AS datetime2(7)) = @rightPartitionBoundary) 
+        BEGIN
+            /* Creates new empty partition by creating new boundary value and specifying NEXT USED file group. */
+            ALTER PARTITION SCHEME PartitionScheme_ResourceChangeData_Timestamp NEXT USED [Primary];
+            ALTER PARTITION FUNCTION PartitionFunction_ResourceChangeData_Timestamp() SPLIT RANGE(@rightPartitionBoundary);
+        END;
+            
+        SET @numberOfHistoryPartitions -= 1;
+    END;
+END;
+GO
+
+EXEC dbo.LogSchemaMigrationProgress 'Creating a staging table for resource change data table.';
```

Comment:
here as well - Creating ResourceChangeDataStaging table

Reply:
Updated the log.